File: scratch/fable_test/Test.ChurchFFICheatcode.fs
[<AutoOpen>]
module PureScript_Test_ChurchFFICheatcode

open System
open System.Collections.Generic

module Test_ChurchFFICheatcode_FFI =
    let runChurchFFICheatcode (input: obj) =
        let count = pown (unbox<int> input) 5
        let mutable result = 0
        for _ in 1 .. count do result <- result + 1
        result :> obj
    

let Test_ChurchFFICheatcode_runChurchFFICheatcode = box (fun (arg0: obj) -> box (Test_ChurchFFICheatcode_FFI.``runChurchFFICheatcode`` (unbox arg0)))


let Test_ChurchFFICheatcode_describe  = (sharpurs_apply (box ((box Effect_Console_log))) (box ((box "Church Numerals FFICheatcode (100k Closure Applications):"))))

let Test_ChurchFFICheatcode_act  = (sharpurs_apply (box ((sharpurs_apply (box ((sharpurs_apply (box ((box Control_Bind_bind))) (box ((box Effect_bindEffect)))))) (box ((sharpurs_apply (box ((box Bench_opaque))) (box ((box 10))))))))) (box ((box (fun (dummy: obj) -> (sharpurs_apply (box ((sharpurs_apply (box ((box Control_Applicative_pure))) (box ((box Effect_applicativeEffect)))))) (box ((sharpurs_apply (box ((sharpurs_apply (box ((box Data_Show_show))) (box ((box Data_Show_showInt)))))) (box ((sharpurs_apply (box ((box Test_ChurchFFICheatcode_runChurchFFICheatcode))) (box ((box dummy)))))))))))))))

File: scratch/fable_test/fable_output/Test.ChurchFFICheatcode.rs
pub mod PureScript_Test_ChurchFFICheatcode {
    use super::*;
    use fable_library_rust::Native_::Any;
    use fable_library_rust::Native_::Func1;
    use fable_library_rust::Native_::MutCell;
    use fable_library_rust::String_::string;
    use crate::module_4da9e9e9::PureScript_Bench;
    use crate::module_ce6bdf8a::PureScript_Control_Applicative;
    use crate::module_6c4d46c3::PureScript_Control_Bind;
    use crate::module_baebcb76::PureScript_Data_Show;
    use crate::module_3ed61c25::PureScript_Effect_Console;
    use crate::module_5706e1fc::PureScript_Effect;
    use crate::module_aa21d1e7::Sharpurs_Prelude;
    pub mod Test_ChurchFFICheatcode_FFI {
        use super::*;
        pub fn runChurchFFICheatcode(input: &dyn Any) -> &dyn Any {
            let count: i32 = Sharpurs_Prelude::unbox(input).pow(5_i32);
            let result: MutCell<i32> = MutCell::new(0_i32);
            for forLoopVar in 1_i32..=count {
                result.set(result.get() + 1_i32);
            }
            &result.get()
        }
    }
    pub fn Test_ChurchFFICheatcode_runChurchFFICheatcode() -> &dyn Any {
        static Test_ChurchFFICheatcode_runChurchFFICheatcode:
         MutCell<Option<&dyn Any>> =
            MutCell::new(None);
        Test_ChurchFFICheatcode_runChurchFFICheatcode.get_or_init(||
                                                                      &Func1::new(move
                                                                                      |arg0|
                                                                                      &PureScript_Test_ChurchFFICheatcode::Test_ChurchFFICheatcode_FFI::runChurchFFICheatcode(&Sharpurs_Prelude::unbox(arg0))))
    }
    pub fn Test_ChurchFFICheatcode_describe() -> &dyn Any {
        static Test_ChurchFFICheatcode_describe: MutCell<Option<&dyn Any>> =
            MutCell::new(None);
        Test_ChurchFFICheatcode_describe.get_or_init(||
                                                         Sharpurs_Prelude::sharpurs_apply(&&&PureScript_Effect_Console::Effect_Console_log(),
                                                                                          &&&string("Church Numerals FFICheatcode (100k Closure Applications):")))
    }
    pub fn Test_ChurchFFICheatcode_act() -> &dyn Any {
        static Test_ChurchFFICheatcode_act: MutCell<Option<&dyn Any>> =
            MutCell::new(None);
        Test_ChurchFFICheatcode_act.get_or_init(||
                                                    Sharpurs_Prelude::sharpurs_apply(&&Sharpurs_Prelude::sharpurs_apply(&&Sharpurs_Prelude::sharpurs_apply(&&&PureScript_Control_Bind::Control_Bind_bind(),
                                                                                                                                                           &&&PureScript_Effect::Effect_bindEffect()),
                                                                                                                        &&Sharpurs_Prelude::sharpurs_apply(&&&PureScript_Bench::Bench_opaque(),
                                                                                                                                                           &&&10_i32)),
                                                                                     &&&Func1::new(move
                                                                                                       |dummy|
                                                                                                       Sharpurs_Prelude::sharpurs_apply(&&Sharpurs_Prelude::sharpurs_apply(&&&PureScript_Control_Applicative::Control_Applicative_pure(),
                                                                                                                                                                           &&&PureScript_Effect::Effect_applicativeEffect()),
                                                                                                                                        &&Sharpurs_Prelude::sharpurs_apply(&&Sharpurs_Prelude::sharpurs_apply(&&&PureScript_Data_Show::Data_Show_show(),
                                                                                                                                                                                                              &&&PureScript_Data_Show::Data_Show_showInt()),
                                                                                                                                                                           &&Sharpurs_Prelude::sharpurs_apply(&&&PureScript_Test_ChurchFFICheatcode::Test_ChurchFFICheatcode_runChurchFFICheatcode(),
                                                                                                                                                                                                              dummy))))))
    }
}
